jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/ServiceHandlerTest.java

```diff
@@ -123,102 +133,76 @@ public void getServicesBothPortAndHealessTest() {
+                .hasFieldOrPropertyWithValue("port", 8080))
+            .satisfies(s -> assertThat(s.getMetadata())
+                .hasFieldOrPropertyWithValue("name", "service-test")
+                .extracting(ObjectMeta::getLabels)
+                .asInstanceOf(InstanceOfAssertFactories.MAP)
+                .containsEntry("expose", "true"));
     }
 
     @Test
-    public void getServicesWithDefaultProtocolTest() {
-        //DEFAULT
+    void getServicesWithNullProtocol() {
         port = ServiceConfig.Port.builder()
                 .port(8080).targetPort(80).nodePort(50).name("port-test").build();
-        ports.clear();
         ports.add(port);
+
         serviceconfig = ServiceConfig.builder()
-                .ports(ports).name("service-test").expose(true).headless(false).type("NodePort").build();
+                .ports(ports).name("service-test").expose(true).headless(false).build();
 
         //creating serviceConfigs of service;
-        serviceConfigs.clear();
         serviceConfigs.add(serviceconfig);
 
-        //calling the getServices Method
         services = serviceHandler.getServices(serviceConfigs);
-        assertEquals("TCP",services.get(0).getSpec().getPorts().get(0).getProtocol());
+        assertThat(services.get(0).getSpec().getType()).isNull();
     }
 
-    @Test
-    public void getServicesWithNullProtocolTest() {
-        //checking null type
+    @ParameterizedTest(name = "With ''{0}'' protocol should return ''{2}''")
+    @MethodSource("data")
+    void getServicesWithProtocol(String protocol, String type, String expectedProtocol) {
         port = ServiceConfig.Port.builder()
-                .port(8080).targetPort(80).nodePort(50).name("port-test").build();
-        ports.clear();
+                .port(8080)
+                .protocol(protocol)
+                .targetPort(80)
+                .nodePort(50)
+                .name("port-test")
+                .build();
         ports.add(port);
 
         serviceconfig = ServiceConfig.builder()
-                .ports(ports).name("service-test").expose(true).headless(false).build();
-
+                .ports(ports)
+                .name("service-test")
+                .expose(true)
+                .headless(false)
+                .type(type)
+                .build();
         //creating serviceConfigs of service;
-        serviceConfigs.clear();
         serviceConfigs.add(serviceconfig);
 
-        //calling the getServices Method
         services = serviceHandler.getServices(serviceConfigs);
-        assertNull(services.get(0).getSpec().getType());
+        assertThat(services.get(0).getSpec().getPorts().get(0).getProtocol()).isEqualTo(expectedProtocol);
+    }
 
+    public static Stream<Arguments> data() {
+        return Stream.of(
+                Arguments.of("tcp", "NodePort", "TCP"),
+                Arguments.of("udp", "NodePort", "UDP"),
+                Arguments.of(null, "NodePort", "TCP")
+        );
```

Comment:
type is always `NodePort` why is this a parameter?

Reply:
Missed due to null and default protocol tests.